libbeat/common/transport/httpcommon/proxy.go

```diff
@@ -24,6 +24,9 @@ import (
 	"github.com/elastic/beats/v7/libbeat/common"
 )
 
+// ProxyURI is a URL used for proxy serialized as a string.
+type ProxyURI url.URL
```

Comment:
In order to make this type more reusable lets add `MarshalJSON` and unmarshallers for json, yml and config loading.

For integration with `go-ucfg` add:

```
func (p *ProxyURI) Unpack(s string) error {
  ...
}
```